org.librarysimplified.r2.vanilla/src/main/assets/scripts/utils.js

```diff
@@ -9,19 +9,32 @@ window.addEventListener("load", function() { // on page load
     } else {
         console.log("last_known_scrollX_position " + last_known_scrollX_position);
     }
+    onReadingPositionChanged()
 }, false);
 
 var last_known_scrollX_position = 0;
 var last_known_scrollY_position = 0;
 var ticking = false;
 
+// Returns the page currently displayed for this chapter as an int [1 - n].
+var getCurrentPage = function() {
+    return Math.abs(window.scrollX / window.innerWidth) + 1;
+}
+
+// Returns the page count for this chapter as an int.
+var getPageCount = function() {
+    return Math.abs(document.scrollingElement.scrollWidth / window.innerWidth);
+}
+
 // Position in range [0 - 1].
-var update = function(position) {
-    let positionString = position.toString();
-    Android.onChapterProgressionChanged(positionString);
+var onReadingPositionChanged = function() {
+    var currentPage = getCurrentPage();
+    var pageCount = getPageCount();
+    console.log("current_page " + currentPage + ", page_count " + pageCount);
+    Android.onReadingPositionChanged(currentPage, pageCount);
 };
 
-function isScrollModeEnabled() {
+var isScrollModeEnabled = function() {
```

Comment:
Is there some reason we're using `var f = function()` instead of `function f() {`? My understanding is that `function f() {`
makes the definition use lexical scoping instead of ... whatever insane scoping rules Javascript had before.

Reply:
I just made this change to make this consistent with the other function definitions in the file. It's just a function expression vs a declaration. I think the big difference is that they're not hoisted like function declarations are.

https://developer.mozilla.org/en-US/docs/web/JavaScript/Reference/Operators/function